File: competitive_programming/max_subArray_sum.cpp

 #include <bits/stdc++.h>
 using namespace std;

 int main(){
    int sum = 0;
    int best = 0;

    vector<int> arr = {-1,2,4,-3,5,2,-5,2};

    for (int i : arr){
        sum = max(i, sum + i);
        best = max(best, sum);
    }

    cout<<best<<"\n";
 }

File: competitive_programming/max_subArray_sum.py

def maxSubAarraySum(array):
    sum_subarray= 0
    best = 0

    for i in array:
        sum_subarray= max(i, i + sum_subarray)
        best = max(best, sum_subarray)

    return best

array = [-1, 2, 4, -3, 5, 2, -5, 2]
print(maxSubAarraySum(array))
